### PR TITLE
Test fixes 2

### PR DESCRIPTION
**Pull request by def-**

### CODE REVIEW

**Comment by oderwat** [comment]
Good that you referenced that in my issue. I was about to do some of that too.

**Comment by def-** [comment]
And I already did more, but at the same time as Araq: https://github.com/Araq/Nim/pull/2306

File: lib/pure/os.nim [hunk 1]
```diff
@@ -1997,6 +1997,8 @@ proc getFileInfo*(handle: FileHandle): FileInfo =
     rawToFormalFileInfo(rawInfo, result)
 
 proc getFileInfo*(file: File): FileInfo =
+  if file.isNil:
```
Comment (dom96) [inline lib/pure/os.nim]: While this helps with debugging does it mean that we should be checking every function param to see whether it is nil?

**Comment by Araq** [comment]
@dom96 I think getFileInfo is special and should do the 'nil' checking.

File: tests/assert/tfailedassert.nim [hunk 2]
```diff
@@ -3,7 +3,7 @@ discard """
 WARNING: false first assertion from bar
 ERROR: false second assertion from bar
 -1
-tfailedassert.nim:27 false assertion from foo
+tests/assert/tfailedassert.nim:27 false assertion from foo
```
Comment (Araq) [inline tests/assert/tfailedassert.nim]: On windows, this really is 'tfailedassert.nim' only, dunno why though.

**Comment by dom96** [comment]
What makes it special?

**Pull request merged**